doc_source/lambda.md

```diff
@@ -39,189 +33,166 @@ license = "YOUR-LICENSE"
+1. A main function that sets up logging and runs your handler function, routing new requests to it\.
 
-1. \(Optional\) A struct that represents data that your lambda returns\. This is typically a `Result` wrapping either a success response or a failure response but can be anything that implements `serde::Serialize`\.
+1. \(Optional\) A struct that represents data that your lambda returns\. This is typically a `Result` wrapping either a success response or a failure response. The success response can be anything that implements `serde::Serialize`\, while the failure response can be anything that implements `std::fmt::Debug + std::fmt::Display`.
 
 Replace `src/main.rs` with the following code\.
 
 ```
+use lambda_runtime::{service_fn, Error, LambdaEvent};
+use serde::{Deserialize, Serialize};
+
 #[derive(Deserialize)]
 struct Request {
-    pub body: String,
+    body: String,
 }
 
 #[derive(Debug, Serialize)]
-struct SuccessResponse {
-    pub body: String,
+struct Response {
+    req_id: String,
+    body: String,
 }
 
-#[derive(Debug, Serialize)]
-struct FailureResponse {
-    pub body: String,
-}
-
-// Implement Display for the Failure response so that we can then implement Error.
-impl std::fmt::Display for FailureResponse {
+impl std::fmt::Display for Response {
+    /// Display the response struct as a JSON string
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.body)
+        let err_as_json = serde_json::json!(self).to_string();
+        write!(f, "{}", err_as_json)
     }
 }
 
-// Implement Error for the FailureResponse so that we can `?` (try) the Response
-// returned by `lambda_runtime::run(func).await` in `fn main`.
-impl std::error::Error for FailureResponse {}
-
-type Response = Result<SuccessResponse, FailureResponse>;
-
-#[tokio::main]
-async fn main() -> Result<(), lambda_runtime::Error> {
-    let func = handler_fn(handler);
-    lambda_runtime::run(func).await?;
+impl std::error::Error for Response {}
 
-    Ok(())
-}
-
-async fn handler(req: Request, _ctx: lambda_runtime::Context) -> Response {
-    info!("handling a request...");
-    let bucket_name = std::env::var("BUCKET_NAME")
-        .expect("A BUCKET_NAME must be set in this app's Lambda environment variables.");
-
-    // No extra configuration is needed as long as your Lambda has
-    // the necessary permissions attached to its role.
-    let config = aws_config::load_from_env().await;
-    let s3_client = aws_sdk_s3::Client::new(&config);
+async fn put_object(
+    s3_client: &aws_sdk_s3::Client,
+    bucket_name: &str,
+    event: LambdaEvent<Request>,
+) -> Result<Response, Error> {
+    tracing::info!("handling a request...");
     // Generate a filename based on when the request was received.
-    let filename = format!("{}.txt", time::OffsetDateTime::now_utc().unix_timestamp());
+    let filename = format!("{}.txt", chrono::offset::Utc::now().timestamp());
```

Comment:
I don't think it's worth pulling in the entire chrono crate for generating a file name. Was std::time not giving the right values for the name?